owncloudDomain/src/main/java/com/owncloud/android/domain/files/model/OCFile.kt

```diff
@@ -23,25 +23,52 @@ package com.owncloud.android.domain.files.model
 import android.os.Parcelable
 import android.webkit.MimeTypeMap
 import kotlinx.android.parcel.Parcelize
+import java.io.File
 import java.util.Locale
 
 //TODO: Add new attributes on demand. Let's try to perform a clean up :)
 @Parcelize
 data class OCFile(
-    val id: Long? = null,
-    val parentId: Long? = null,
+    var id: Long? = null,
+    var parentId: Long? = null,
     val owner: String,
-    val length: Long,
-    val creationTimestamp: Long? = null,
-    val modifiedTimestamp: Long,
+    var length: Long,
+    var creationTimestamp: Long? = 0,
+    var modificationTimestamp: Long,
     val remotePath: String,
-    val mimeType: String,
-    val etag: String? = null,
+    var mimeType: String,
+    var etag: String? = "",
     val permissions: String? = null,
-    val remoteId: String? = null,
-    val privateLink: String? = null
+    var remoteId: String? = null,
+    val privateLink: String? = "",
+    var storagePath: String? = null,
+    var treeEtag: String? = "",
+
+    // May not needed
+    val keepInSync: Int? = null,
+    var lastSyncDateForData: Int? = 0,
+    var lastSyncDateForProperties: Long? = 0,
+    var needsToUpdateThumbnail: Boolean = false,
+    var modifiedAtLastSyncForData: Int? = 0,
+    var etagInConflict: String? = null,
+    val fileIsDownloading: Boolean? = false,
+    var sharedWithSharee: Boolean? = false,
+    var sharedByLink: Boolean = false
 ) : Parcelable {
 
+    val fileName: String
+        get() = File(remotePath).name.let { if (it.isBlank()) ROOT_PATH else it }
+
+    @Deprecated("Do not use this constructor. Remove it as soon as possible")
+    constructor(remotePath: String, mimeType: String, parentId: Long?, owner: String) : this(
```

Comment:
Because i need to create a new ocfile from java files. As soon as we move those files to kotlin, i will remove that constructor.

Reply:
But can't you call the default constructor from java?

Reply:
No, because nullable and not nullable attributes are intercalated. I don't like to create a constructor for this, but the plan is to remove as soon as possible once we move forward more new_arch stuff. This will be merged to new_arch/synchronization and there is still a lot of work to do before merging it to master.